subsys/bluetooth/controller/crypto/crypto.c

```diff
@@ -5,28 +5,24 @@
  */
 
 #include <soc.h>
+#include <entropy.h>
 
 #define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
 #include "common/log.h"
 
 #include "hal/cpu.h"
-#include "hal/rand.h"
 #include "hal/ecb.h"
 
-K_MUTEX_DEFINE(mutex_rand);
-
 int bt_rand(void *buf, size_t len)
 {
-	while (len) {
-		k_mutex_lock(&mutex_rand, K_FOREVER);
-		len = rand_get(len, buf);
-		k_mutex_unlock(&mutex_rand);
-		if (len) {
-			cpu_sleep();
-		}
+	struct device *entropy;
+
+	entropy = device_get_binding(CONFIG_ENTROPY_NAME);
```

Comment:
Can this either use the entropy handle from the radio, or is there somewhere we can initialize this once, so we don't have to call device_get_binding on every call to bt_rand()

Reply:
I plan to change bt_rand later anyway, and this is only used by the BLE Host sparingly.